pkg/bucket/bandwidth/token/token.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package token
+
+import (
+	"fmt"
+	"sync"
+)
+
+// NameTokenizer maps names to tokens.
```

Comment:
Not sure I see the need for this. The Reader can just hold a reference to its parent through the channel.

If you need to track inflight objects just have an StartObject and StopObject.